PersonIO&Threads/src/PersonRepository.java

```diff
@@ -0,0 +1,99 @@
+import java.io.*;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class PersonRepository {
+
+    private static List<Person> personList = new ArrayList<>();
+    private static List<Person> birthdayGirls = new ArrayList<>();
+
+    /**
+     * This method reads a file, parses its content, instantiates Person objects and adds them to a list.
+     *
+     * @param filename
+     */
+    public void readFromFileToList(String filename) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(filename))) {
+            String s;
+            while ((s = reader.readLine()) != null) {
+                if ((s.length() > 0)) {
+                    String[] stringReader = s.split(",");
+                    Person p = new Person((stringReader[0]), stringReader[1]);
+                    p.setGender(stringReader[2]);
+                    addPerson(p);
```

Comment:
Ca sa nu faci de fiecare data cand creezi o persoana add in `personList` cu aceasta metoda `synchronised`. Puteai sa adaugi obiectele intro lista locala in aceasta metoda, iar apoi sa adaugi toate acele obiecte odata in `personList`. Astfel eviti acele multe apeluri la metoda `synchronised` care iti pot incetini programul.